public/docs/_examples/server-communication/dart/lib/toh/hero_service.dart

```diff
@@ -0,0 +1,45 @@
+// #docplaster
+
+// #docregion
+import 'package:angular2/angular2.dart';
+import 'dart:convert';
+import 'dart:async';
+// #enddocregion v1
+// #docregion import-request-options
+import 'package:http/browser_client.dart';
+// #enddocregion import-request-options
+// #docregion v1
+import 'hero.dart';
+
+@Injectable()
+class HeroService {
+  String _heroesUrl = 'app/heroes';
+
+  BrowserClient _http;
+
+  HeroService(this._http);
+
+// #docregion methods
+  Future<List<Hero>> getHeroes() async {
+    final response = await _http.get(_heroesUrl);
+    Map data = JSON.decode(response.body);
+    List<Hero> heroes = _toHeroes(data['data']);
```

Comment:
I think it would be cleaner (and closer to the ts version) to do something like this:
  var heroes = JSON.decode(response.body)['data'].map((value) => new Hero.fromJson(value));